src/model.rs

```diff
@@ -53,47 +40,79 @@ define_language! {
     }
 }
 
-pub struct TensorAnalysis {
-  pub graph: std::cell::RefCell<Box<Graph>>
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum DataKind {
+  Name,
+  Scalar,
+  Tnsr,
 }
 
-impl Default for TensorAnalysis {
-  fn default() -> Self {
-    unsafe {
-      // NOTE Box heap-allocates, otherwise any pointer from
-      // C++ may be dangling
-      let mut graph = Box::new(Graph::new());
-      Graph_Graph(&mut *graph);
-      TensorAnalysis { graph: std::cell::RefCell::new(graph) }
-    } 
-  } 
+
+impl Default for DataKind {
+    fn default() -> Self {
+        DataKind::Name
+    }
 }
 
+/// Metadata struct for TensorAnalysis
+///
+/// # Fields
+///
+/// - `dtype`: The data type of this eclass, can be a name/scalar/tensor
```

Comment:
Move these into the struct. Comment above each field.

Reply:
I looked up some rust comment style guide, and seems that Rust convention is to have documentations above struct/function: https://deterministic.space/machine-readable-inline-markdown-code-cocumentation.html. Also checked egg: https://github.com/mwillsey/egg. Perhaps it's some special convention for Rust...

Reply:
I think the template you use is for function. See: https://doc.rust-lang.org/stable/rust-by-example/meta/doc.html

Reply:
I see! Just added a commit that changes it. Thanks!